firefox-ios/Client/Frontend/Browser/SearchEngines/SearchEngines.swift

```diff
@@ -192,12 +192,15 @@ class SearchEngines {
         }
     }
 
+    var shouldHidePrivateModeFirefoxSuggestSetting: Bool {
+        return !shouldShowBookmarksSuggestions &&
+        !shouldShowSyncedTabsSuggestions &&
+        !shouldShowBrowsingHistorySuggestions
+    }
+
```

Comment:
This logic seems to be related to Firefox Suggest, but lives in Search Engine class (which imo is separate). The `SearchSettingsTableViewController` seems to be a bit large, do we see a benefit in creating a viewmodel to hold all these boolean values instead? Just consideration, not a blocker on this PR.

Reply:
I'll relocate the logic to the SearchSettingsTableViewController, and I'll establish the view model in another PR, migrating all necessary logic there. Time constraints are necessitating us to accelerate this process.